components/dashboard/src/repositories/detail/workspaces/WorkpaceSizeOptions.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import type { Configuration } from "@gitpod/public-api/lib/gitpod/v1/configuration_pb";
+import React, { useCallback, useState } from "react";
+import { useWorkspaceClasses } from "../../../data/workspaces/workspace-classes-query";
+import { Label } from "@podkit/forms/Label";
+import { RadioGroup, RadioGroupItem } from "@podkit/forms/RadioListField";
+import { TextMuted } from "@podkit/typography/TextMuted";
+import { Heading2 } from "@podkit/typography/Headings";
+import { ConfigurationSettingsField } from "../ConfigurationSettingsField";
+import { useUpdateProject } from "../../../data/projects/project-queries";
+import { useToast } from "../../../components/toasts/Toasts";
+import { LoadingButton } from "@podkit/buttons/LoadingButton";
+
+interface Props {
+    configuration: Configuration;
+}
+
+export const ConfigurationWorkspaceSizeOptions = ({ configuration }: Props) => {
+    const [selectedValue, setSelectedValue] = useState(
+        configuration.workspaceSettings?.workspaceClass || "g1-standard",
+    );
+    const classChanged = selectedValue !== configuration.workspaceSettings?.workspaceClass;
+
+    const updateProject = useUpdateProject();
+    const { data: classes, isError, isLoading } = useWorkspaceClasses();
+
+    const { toast } = useToast();
+
+    const setWorkspaceClass = useCallback(
+        async (e: React.FormEvent) => {
+            e.preventDefault();
+
+            updateProject.mutate(
+                {
+                    id: configuration.id,
+                    settings: {
+                        workspaceClasses: {
+                            regular: selectedValue,
+                        },
+                    },
+                },
+                {
+                    onSuccess: () => {
+                        // todo: use optimistic updates when we introduce configuration update hooks
+                        toast({ message: "Workspace size updated" });
+                    },
+                    onError: (e) => {
+                        toast({ message: `Failed updating workspace size: ${e.message}` });
+                    },
+                },
+            );
+        },
+        [configuration.id, selectedValue, toast, updateProject],
+    );
+
+    if (isError) {
+        return <div>Something went wrong</div>;
+    }
+
+    if (isLoading || !classes) {
+        return <div>Loading...</div>;
```

Comment:
Would be nice to flush this out a little more. Some thoughts:

1. Loading state. I think if we did the following and tossed it into a `<LoadingState />` (or a better name 😄 ) component in podkit. This will render a spinner, but delay rendering by 1 second so we avoid the spinner flash most of the time, but render if it's taking longer.

```tsx
// This exists in our component lib, we could lift it up into podkit
<Delayed>
  <Loader2 className="animate-spin" />
</Delayed>
```

2. `!classes` - If we're not loading anymore and there are no classes, I think we should treat this as an error state.
3. Error state - Let's render an error alert of some sort (maybe we need Cory's help on a generic error alert (could be similar to what we have today?)

I don't think these are blocking so feel free to decide what makes since to do now vs. followups.

Reply:
I like the loading state, will implement the comp.